air/src/air/logup_gkr/mod.rs

```diff
@@ -229,3 +250,72 @@ pub enum LogUpGkrOracle<B: StarkField> {
     /// must be a power of 2.
     PeriodicValue(Vec<B>),
 }
+
+// PERIODIC COLUMNS FOR LOGUP
+// =================================================================================================
+
+/// Stores the periodic columns used in a LogUp-GKR statement.
+///
+/// Each stored periodic column is interpreted as a multi-linear extension polynomial of the column
+/// with the given periodic values. Due to the periodic nature of the values, storing, binding of
+/// an argument and evaluating the said multi-linear extension can be all done linearly in the size
+/// of the smallest cycle defining the periodic values. Hence we only store the values of this
+/// smallest cycle. The cycle is assumed throughout to be a power of 2.
+#[derive(Clone, Debug, Default, PartialEq, PartialOrd, Eq, Ord)]
+pub struct PeriodicTable<E: FieldElement> {
+    pub table: Vec<Vec<E>>,
+}
+
+impl<E> PeriodicTable<E>
+where
+    E: FieldElement,
+{
+    pub fn new(table: Vec<Vec<E::BaseField>>) -> Self {
+        let mut result = vec![];
+        for col in table.iter() {
+            let res = embed_in_extension(col.to_vec());
+            result.push(res)
+        }
+
+        Self { table: result }
+    }
+
+    pub fn num_columns(&self) -> usize {
+        self.table.len()
+    }
+
+    pub fn table(&self) -> &[Vec<E>] {
+        &self.table
+    }
+
+    pub fn get_periodic_values_at(&self, row: usize, values: &mut [E]) {
+        self.table
+            .iter()
+            .zip(values.iter_mut())
+            .for_each(|(col, value)| *value = col[row % col.len()])
+    }
```

Comment:
nit: let's rename this to `fill_periodic_values_at()`.

Reply:
Changed